packages/eslint-plugin-react-hooks/src/RulesOfHooks.js

```diff
@@ -272,7 +272,7 @@ export default {
           if (pathList.has(segment.id)) {
             const pathArray = Array.from(pathList);
             const cyclicSegments = pathArray.slice(
-              pathArray.indexOf(segment.id) + 1,
+              pathArray.indexOf(segment.id) - 1,
```

Comment:
Question here: `slice(pathArray.indexOf(segment.id))` makes sense -- we want to include the cyclic segment itself when adding to `cyclic`. What's the logic for also including the previous segment in pathsHistory?

I do see that this test case doesn't error, just trying to understand. Could we also add this to ESLintRulesOfHooks-test to be complete?
```js
function ComponentWithHookInsideLoop() {
  do {
	  foo();
  } while (useHookInsideLoop());
}
```

Reply:
Thanks for the test - it covers precisely why we want to include the previous segment. 

I've added the test in https://github.com/facebook/react/pull/28714/commits/dc7a41f13caf21c907a3ff089e68974d6245e05f.

Reply:
i'm not 100% sure, but i think this may have introduced some false positives for other use-cases:
https://github.com/facebook/react/issues/31687#issuecomment-2529150548